controllers/metal3.io/host_state_machine_test.go

```diff
@@ -221,6 +217,20 @@ func TestErrorCountIncreasedWhenProvisionerFails(t *testing.T) {
 	}
 }
 
+func TestErrorCountIncreasedWhenRegistrationFails(t *testing.T) {
```

Comment:
I'm a little surprised at how few changes there are to the tests. Maybe that says more about our test coverage than this change, though.